addons/web/static/src/scss/form_view.scss

```diff
@@ -166,6 +166,18 @@ $o-form-label-margin-right: 0px;
                     width: auto!important;
                 }
             }
+            // in editable form view, display reference field components(input and selection) full-width and inline
```

Comment:
did you try to change line 164 to
```scss
&.o_field_boolean, &.o_priority, &.o_field_many2one_selection {
```
I think it fixes the issue. It is simpler, and it follows the logic already in place (that is, defining a default behavior and overriding it for some fields)